hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java

```diff
@@ -233,11 +239,10 @@ public MetricsContext getMetricsContext() {
     private void stateMachineStep() {
         switch (state) {
```

Comment:
This switch statement is a bit of a monster... Extracting some methods comes to mind, but I can imagine that actually hurting readability. Give it a try, if you feel like it?